python/ray/train/_internal/storage.py

```diff
@@ -429,46 +421,31 @@ def __init__(
     ):
         self.custom_fs_provided = storage_filesystem is not None
 
-        self.storage_local_path = _get_defaults_results_dir()
-
         # If no remote path is set, try to get Ray Storage URI
         ray_storage_uri: Optional[str] = _get_storage_uri()
-        if ray_storage_uri and storage_path is None:
+        if ray_storage_uri is not None:
             logger.info(
                 "Using configured Ray Storage URI as the `storage_path`: "
                 f"{ray_storage_uri}"
             )
+            storage_path = ray_storage_uri
```

Comment:
Should we continue supporting Ray Stroage? Previously, we fall back to ray storage if `storage_path` is not set.

Now, I'm trying to simplify `storage_path` so that it's always set in the `RunConfig` directly -- defaulting to `~/ray_results`. 

This means that it'll never be `None` / auto-resolved anymore. So, the behavior here has changed to: if ray storage is setup, then override the `storage_path`. Thoughts on this change? Or alternatively we can drop this behavior entirely?

Reply:
Ah, I thought of a solution -- we can do this check directly in the `RunConfig` init.

Reply:
`ray_storage_uri` is set through `ray.init(..., storage="s3://...")`, so users will have already called ray init.